cmd/skaffold/app/cmd/runner.go

```diff
@@ -39,58 +40,64 @@ import (
 var createRunner = createNewRunner
 
 func withRunner(ctx context.Context, action func(runner.Runner, *latest.SkaffoldConfig) error) error {
-	runner, config, err := createRunner(opts)
+	runner, config, runCtx, err := createRunner(opts)
 	if err != nil {
 		return err
 	}
-
+	event.InitializeState(runCtx.Cfg, runCtx.KubeContext)
+	event.LogMetaEvent()
 	err = action(runner, config)
-
-	return alwaysSucceedWhenCancelled(ctx, err)
+	if err := alwaysSucceedWhenCancelled(ctx, err); err != nil {
```

Comment:
Optional: I found this function name to be confusing.  The function seems simple enough that it might be better to just inline it here.

Reply:
i am not doing to change this in this PR.